### PR TITLE
refactor(stream): remove unnecessary `Arc<Mutex>` in rearranged_chain

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Merge or not merge, both OK.

~~Just play an example to show how to avoid some unnecessary `Arc<Mutex>` in critical path, but I guess here is not a critical path :)~~

Just use &mut is OK here, blame BUGEN

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Review by skyzh — APPROVED** [review]
🥵🥵🤤🤤

**Review by wangrunji0408 — APPROVED** [review]
🥵🥵😇😇

**Comment by yuhao-su** [comment]
Rust magic😇

File: src/stream/src/executor/rearranged_chain.rs [hunk 1]
```diff
@@ -235,9 +238,8 @@ impl RearrangedChainExecutor {
                 yield msg;
             }
 
-            // Now we take back the remaining upstream. There should be no contention since
-            // `rearranged` stream is already dropped.
-            let mut remaining_upstream = upstream.try_lock().unwrap();
+            // Now we take back the remaining upstream.
+            let remaining_upstream = owner.rw(&upstream2);
```
Comment (BugenZhao) [inline src/stream/src/executor/rearranged_chain.rs:242]: Looks like `Rc`s must be joined before dropping?

**Comment by TennyZhuang** [comment]
Finally, I found that there is no interior mutability required, not sure why Bugen use `Arc<Mutex>` here :)

**Comment by TennyZhuang** [comment]
But at least I've learned ghost_cell in the meaningless try.

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**